### ISSUE TITLE
Incorrect behaviour when max length is less than marker text length, but the string fits.

### ISSUE THREAD

**Issue by mschout** [perlancar/perl-String-Elide-Tiny#1]
Discovered bug where this does the wrong thing:

actual behaviour:
```
$ perl -MString::Elide::Tiny=elide -E "say elide('ab', 2, { truncate => 'right', marker => '...' })"
..
```

So it prints `..` but the string `ab` actually *does* fit, so we should have gotten `ab`

Quick skim this line is the problem:

```
    return substr($marker, 0, $max_len) if $max_len < $marker_len;
```

One possible fix is to just move this line up:

![image](https://github.com/perlancar/perl-String-Elide-Tiny/assets/23033/78fe9cd9-e5e4-4c60-8fee-3d350e5d860c)

IE bail out earlier with `$str` return value if it fits.

**Comment by mschout** [comment]
I'd make a PR, but seems like this repo is out of date with what is actually on CPAN

**Comment by perlancar** [comment]
Hi Michael, thanks for the catch. Added a fix and a couple of other checks, released v0.003.
